corediff.go

```diff
@@ -45,25 +60,34 @@ func loadDB(path string) hashDB {
 }
 
 func saveDB(path string, db hashDB) {
-	f, err := os.Create(path)
+	tmpPath := path + ".tmp"
+	f, err := os.Create(tmpPath)
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer f.Close()
+	defer os.Remove(tmpPath)
 	for k := range db {
 		if err := binary.Write(f, binary.LittleEndian, k); err != nil {
 			log.Fatal(err)
 		}
 	}
+	if err := f.Close(); err != nil {
+		log.Fatal(err)
+	}
+	if err := os.Rename(tmpPath, path); err != nil {
+		log.Fatal(err)
+	}
 }
 
-func parseFile(path, _ string, db hashDB, updateDB bool) (hits []int, lines [][]byte) {
+func parseFile(path string, db hashDB, updateDB bool) (hits []int, lines [][]byte) {
 	fh, err := os.Open(path)
-	if os.IsNotExist(err) {
+	if err != nil && os.IsNotExist(err) {
 		logInfo(warn("file does not exist: " + path))
 		return nil, nil
+	} else if err != nil {
+		log.Fatal("open error on", path, err)
 	}
```

Comment:
Possibly:

```suggestion
    if err != nil {
		log.Fatal("open error on", path, err)
	}
	if os.IsNotExist(err) {
		logInfo(warn("file does not exist: " + path))
		return nil, nil
	}
```

Reply:
your suggestion would not reach line 89

Reply:
as log.Fatal does an os.Exit

Reply:
Thanks, I see. I had a feeling that the condition can be simplified somehow, but it seems no - let's keep it the way it is then